utils/test_utils/src/chain_structures.rs

```diff
@@ -189,3 +189,21 @@ pub fn construct_chain_exchange_response() -> ChainExchangeResponse {
         message: "message".to_owned(),
     }
 }
+
+// Serialize macro used for testing
+#[macro_export]
+macro_rules! to_string_with {
+    ($obj:expr, $serializer:path) => {{
+        let mut writer = Vec::new();
+        $serializer($obj, &mut serde_json::ser::Serializer::new(&mut writer)).unwrap();
+        String::from_utf8(writer).unwrap()
+    }};
+}
+
+// Deserialize macro used for testing
+#[macro_export]
+macro_rules! from_str_with {
+    ($str:expr, $deserializer:path) => {
+        $deserializer(&mut serde_json::de::Deserializer::from_str($str)).unwrap()
+    };
+}
```

Comment:
I'm not sure `chain_structures.rs` is the right home. How about you put them in `lib.rs` for now?